File: 70-climbing-stairs/70-climbing-stairs.scala
object Solution {
    def climbStairs(n: Int): Int = {
        val arr: Array[Int] = Array.fill(n + 1) {0}
        arr(0) = 1
        arr(1) = 1
        
        for(i <- 2 to n) {
            arr(i) = arr(i - 1) + arr(i - 2)
        }
        
        arr(n)
    }
}

File: 70-climbing-stairs/70-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        int n1 = 1;
        int n2 = 2;
        
        if (n == 1) return 1;
        
        for(int i = 3; i <= n; i++) {
            int c = n1 + n2;
            n1 = n2;
            n2 = c;
        }
        
        return n2;
    }
}